py-polars/polars/internals/expr/expr.py

```diff
@@ -3396,10 +3396,17 @@ def get_lazy_promise(df: pli.DataFrame) -> pli.LazyFrame:
 
                 # create partitions with LazyFrames
                 # these are promises on a computation
-                partitions = [
-                    get_lazy_promise(df[i * chunk_size : i * chunk_size + chunk_size])
-                    for i in range(n_threads)
-                ]
+                partitions = []
+                for i in range(n_threads):
+                    # latest partition can be a little longer
+                    # we increase length of slice
+                    if i == (n_threads - 1):
+                        partition_df = df[
+                            i * chunk_size : i * chunk_size + chunk_size * 10
```

Comment:
But why by the factor of 10?

Reply:
Doesn't really matter. 2 would suffice as well I think.

Reply:
It's because it's truncated, so the last chunk will be size `[chunk_size; chunk_size * 2)`? Right.

Reply:
I don't think it will be exactly that size. It was missing 2 rows in the failing test. It is just because of rounding errors in floor division.

Reply:
In any case. It doesn't really matter as long as we fill in a length that is longer than the actual length.

Reply:
It's obviously irrelevant for this purpose, but I suppose the "ideal" way of dividing up the chunks looks like this.
```python
x_len = 50
n_threads = 6
chunk_size = x_len // n_threads
remainder = x_len % n_threads

a = 0
b = chunk_size
for i in range(n_threads):
    if i >= n_threads - remainder:
        b += 1
    print(f"{a:02d} - {b:02d}, (len = {b-a})")
    a = b
    b += chunk_size
    
"""
00 - 08, (len = 8)
08 - 16, (len = 8)
16 - 24, (len = 8)
24 - 32, (len = 8)
32 - 41, (len = 9)
41 - 50, (len = 9)
""" 
```
😸

Reply:
Uhh does the code work when the number of cores is bigger than the number of rows?

Reply:
I think what you actually wanted is `df[i * chunk_size : i * chunk_size + chunk_size + remainder]`.
You're assuming that `remainder <= 9 * chunk_size`.  
This fails, for example when `len_data = 9` and `n_cores = 10`.

On my computer, the following results in a `ComputeError`.
```python
import os
os.environ["POLARS_MAX_THREADS"] = "10"
import polars as pl

n = pl.threadpool_size()
df = pl.DataFrame({"x": [1 for _ in range(n - 1)]})
df.select(pl.col("x").apply(lambda t: 0, strategy="threading"))
```